firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryRemoteDocumentCache.java

```diff
@@ -86,4 +86,28 @@ public MaybeDocument get(DocumentKey key) {
   ImmutableSortedMap<DocumentKey, MaybeDocument> getDocuments() {
     return docs;
   }
+
+  /**
+   * Returns an estimate of the number of bytes used to store the given document key in memory. This
+   * is only an estimate and includes the size of the segments of the path, but not any object
+   * overhead or path separators.
+   */
+  private static long getKeySize(DocumentKey key) {
+    ResourcePath path = key.getPath();
+    long count = 0;
+    for (int i = 0; i < path.length(); i++) {
+      // Strings in java are utf-16, each character is two bytes in memory
+      count += path.getSegment(i).length() * 2;
+    }
+    return count;
```

Comment:
I am not 100% sold that this is meaningful in approximating our cache size. On top of that, there is also a lot of overhead that we are not accounting for ("a/b/c" is more expensive in terms of storage than "abc"). 

On the other hand, you did already write this code so maybe we can keep it.